src/main/java/com/timgroup/statsd/NonBlockingStatsDClient.java

```diff
@@ -210,389 +135,34 @@ public void stop() {
-     */
-    @Override
-    public void recordEvent(final Event event, final String... tags) {
-        final String title = prefix + event.getTitle();
-        final String text = event.getText();
-        send(String.format("_e{%d,%d}:%s|%s%s%s",
-                title.length(), text.length(), title, text, eventMap(event), tagString(tags)));
-    }
-
-    /**
-     * Records a run status for the specified named service check.
-     *
-     * <p>This method is a DataDog extension, and may not work with other servers.</p>
-     *
-     * <p>This method is non-blocking and is guaranteed not to throw an exception.</p>
-     *
-     * @param sc
-     *     the service check object
-     */
-    @Override
-    public void recordServiceCheckRun(final ServiceCheck sc) {
-        send(toStatsDString(sc));
     }
     
-    private String toStatsDString(final ServiceCheck sc) {
-        // see http://docs.datadoghq.com/guides/dogstatsd/#service-checks
-        final StringBuilder sb = new StringBuilder();
-        sb.append(String.format("_sc|%s|%d", sc.getName(), sc.getStatus()));
-        if (sc.getTimestamp() > 0) {
-            sb.append(String.format("|d:%d", sc.getTimestamp()));
-        }
-        if (sc.getHostname() != null) {
-            sb.append(String.format("|h:%s", sc.getHostname()));
-        }
-        sb.append(tagString(sc.getTags()));
-        if (sc.getMessage() != null) {
-            sb.append(String.format("|m:%s", sc.getEscapedMessage()));
-        }
-        return sb.toString();
-    }
-
-    /**
-     * Convenience method equivalent to {@link #recordServiceCheckRun(ServiceCheck sc)}.
-     */
-    @Override
-    public void serviceCheck(final ServiceCheck sc) {
-        recordServiceCheckRun(sc);
-    }
-
-    private void send(String message) {
+    protected void send(String message) {
         queue.offer(message);
     }
+    
+    private void blockingSend(String message) {
+        super.send(message);
+    }
 
     public static final Charset MESSAGE_CHARSET = Charset.forName("UTF-8");
 
     private class QueueConsumer implements Runnable {
-        private final ByteBuffer sendBuffer = ByteBuffer.allocate(PACKET_SIZE_BYTES);
-
         @Override public void run() {
             while(!executor.isShutdown()) {
                 try {
-                    String message = queue.poll(1, TimeUnit.SECONDS);
-                    if(null != message) {
-                        byte[] data = message.getBytes(MESSAGE_CHARSET);
-                        if(sendBuffer.remaining() < (data.length + 1)) {
-                            blockingSend();
-                        }
-                        if(sendBuffer.position() > 0) {
-                            sendBuffer.put( (byte) '\n');
-                        }
-                        sendBuffer.put(data);
-                        if(null == queue.peek()) {
-                            blockingSend();
-                        }
+                    if (queue.peek() == null) {
```

Comment:
If you'd like me to just duplicate the code, that's no problem. I figured we would want to have a consistent implementation between the two, and I don't see any performance issues (in fact, I'm pretty sure I increased performance), but it's your repo and I want to help any way I can.